src/Adapter/Console/ContextLoader.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Console;
+
+use Employee;
+use ConsoleController;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Exception\LogicException;
+use PrestaShop\PrestaShop\Adapter\Shop\Context as ShopContext;
+use PrestaShop\PrestaShop\Core\Console\ContextLoaderInterface;
+
+/**
+ * PrestaShop Context in Console Application
+ */
+final class ContextLoader implements ContextLoaderInterface
+{
+    private $legacyContext;
+    private $shopContext;
+
+    public function __construct(LegacyContext $legacyContext, ShopContext $shopContext, $rootDir)
+    {
+        $this->legacyContext = $legacyContext;
+        $this->shopContext = $shopContext;
+
+        require_once $rootDir . '/../config/config.inc.php';
```

Comment:
Could be used for a LFI, please add a validation here

Reply:
you mean, validate that $rootDir will be a folder? in all cases, this file path will be dynamic.

Reply:
Maybe better to change this line https://github.com/PrestaShop/PrestaShop/blob/develop/autoload.php#L31 in config.inc.php to load the legacy context because of this https://github.com/PrestaShop/PrestaShop/blob/develop/bin/console#L16

Reply:
Hello @PierreRambaud, I see what you mean but IMHO the comment on the related file ("Allow call of Legacy classes from classes in /src and /tests") doesn't work as expected.

Reply:
This is why it could be interesting to change the line and make it works

Reply:
I'm not sure of the impacts as this file is also used in all contexts, I'd razer not rely on it and focus on what we need when we are in Console context.

How about a new file named autoload.php in console folder instead ?

Reply:
This file is only loaded in `admin-dev/index.php` and `./bin/console`.
No need to create a new one to do the job its must do

Reply:
that the thing: I don't think we should use this file at all in Console context: I will remove this call and see what happens :)